k8s/migratev2/scripts/funcs.sh

```diff
@@ -42,3 +43,12 @@ sanity () {
   fi
   echo "==================================="
 }
+
+trim () {
+  local var="$*"
+  # remove leading whitespace characters
+  var="${var#"${var%%[![:space:]]*}"}"
```

Comment:
neat.  the bash substitutions here are novel to me..


`[![:space:]]*` and `*[![:space:]]` distinguish leading and trailing space, `${var##<glob>}` and `${var##<glob>}` produce a matchs of a given (extended) globs, and 


`${var#<glob>}`, `${var%<glob>}` remove leading and trailing and globs, respectively

```
bash-3.2$ myvar='zxcvzxvc'
bash-3.2$ echo ${myvar#'zxc'}
vzxvc
bash-3.2$ echo ${myvar%'xvc'}
zxcvz
bash-3.2$ var='   zxcvzxvc    '
bash-3.2$ echo "'""${var%xvc*}""'"
'   zxcvz'
bash-3.2$ echo "'""${var#xvc*}""'"
'   zxcvzxvc    '
```

a little slow on the uptake on what distinguishes the two `%` or `#`s forms, although i did verify that they are distinct

```
bash-3.2$ echo "'""${var%%[![:space:]]*}""'"
'   '
bash-3.2$ echo "'""${var%[![:space:]]*}""'"
'   zxcvzxv'
```

anyway, i'm sure it's described in https://www.tldp.org/LDP/abs/html/ somewhere